src/DotNetWorker.Grpc/GrpcWorker.cs

```diff
@@ -56,12 +59,26 @@ internal class GrpcWorker : IWorker
             _startupOptions = startupOptions ?? throw new ArgumentNullException(nameof(startupOptions));
             _serializer = workerOptions.Value.Serializer ?? throw new InvalidOperationException(nameof(workerOptions.Value.Serializer));
             _inputConversionFeatureProvider = inputConversionFeatureProvider ?? throw new ArgumentNullException(nameof(inputConversionFeatureProvider));
+            _functionMetadataProvider = functionMetadataProvider ?? throw new ArgumentNullException(nameof(functionMetadataProvider));
         }
 
         public async Task StartAsync(CancellationToken token)
         {
             var eventStream = _rpcClient.EventStream(cancellationToken: token);
 
+            if(_startupOptions.Value.ScriptRoot is null)
+            {
+                throw new ArgumentNullException(nameof(_startupOptions.Value.ScriptRoot));
+            }
+
+            if (!string.IsNullOrEmpty(_startupOptions.Value.AzureWebJobsFeatureFlags))
```

Comment:
Adding this here to make sure duplicate worker-indexing isn't being done. Previously, worker-indexing would be performed here no matter the settings on the host/worker, but it wouldn't break anything b/c this info wasn't sent to the host unless asked for. 

Setting a check to only perform this work if we know that the host will try to ask for it (aka it has EnableWorkerIndexing set). We could additionally make this check stricter by checking that worker.config.json had "workerIndexing" set to true as well, but I think this is check is good enough. Any thoughts?